### DISCUSSION TITLE
How can I access data of included table with an explicit select?

### DISCUSSION THREAD

**Discussion by blastiko** [wheels-dev/wheels#1220] [General]
Hi guys

CFWheels 2.4.0
MySql 5.7.38

I don't know how to access data of included table when i explicit fields in select clause.

I have two table

"buildings" table
CREATE TABLE `buildings` (
	`id` INT(11) NOT NULL AUTO_INCREMENT,
	`name` VARCHAR(250) NOT NULL COLLATE 'utf8_general_ci',
	`address` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`city` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`provinceid` INT(11) NULL DEFAULT NULL,
	`phone` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`email` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`referent` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`note` TEXT NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`noteprice` TEXT NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`district` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`public_transport` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`options` TEXT NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`contact_date` DATETIME NULL DEFAULT NULL,
	`url` VARCHAR(250) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`latitude` VARCHAR(75) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`longitude` VARCHAR(75) NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`createdat` DATETIME NULL DEFAULT NULL,
	`updatedat` DATETIME NULL DEFAULT NULL,
	`deletedat` DATETIME NULL DEFAULT NULL,
	PRIMARY KEY (`id`) USING BTREE,
	INDEX `fk_buildings_provinces1_idx` (`provinceid`) USING BTREE,
	INDEX `city_idx` (`city`) USING BTREE,
	CONSTRAINT `fk_buildings_provinces1` FOREIGN KEY (`provinceid`) REFERENCES `provinces` (`id`) ON UPDATE CASCADE ON DELETE SET NULL
)
COLLATE='utf8_general_ci'
ENGINE=InnoDB
AUTO_INCREMENT=225
;

"classrooms" table
CREATE TABLE `classrooms` (
	`id` INT(11) NOT NULL AUTO_INCREMENT,
	`name` VARCHAR(250) NOT NULL COLLATE 'utf8_general_ci',
	`buildingid` INT(11) NOT NULL,
	`seats` INT(11) NULL DEFAULT NULL,
	`size` INT(11) NULL DEFAULT NULL COMMENT 'mq',
	`price` DECIMAL(10,2) NULL DEFAULT '0.00',
	`note` TEXT NULL DEFAULT NULL COLLATE 'utf8_general_ci',
	`createdat` DATETIME NULL DEFAULT NULL,
	`updatedat` DATETIME NULL DEFAULT NULL,
	`deletedat` DATETIME NULL DEFAULT NULL,
	PRIMARY KEY (`id`) USING BTREE,
	INDEX `fk_masters_locations1_idx` (`buildingid`) USING BTREE,
	INDEX `seats_idx` (`seats`) USING BTREE,
	INDEX `size_idx` (`size`) USING BTREE,
	CONSTRAINT `fk_masters_locations1` FOREIGN KEY (`buildingid`) REFERENCES `buildings` (`id`) ON UPDATE CASCADE ON DELETE CASCADE
)
COLLATE='utf8_general_ci'
ENGINE=InnoDB
AUTO_INCREMENT=329
;

This are models

Building.cfc
component extends="Model" {
	function config() {
		belongsTo(name = "province");
		hasMany(name = "classrooms");
		hasMany(name = "buildinglocations");
		validatesPresenceOf(properties="name", message="Il nome della Sede è abbligatorio");
	}
}
Classroom.cfc
component extends="Model" {
	function config() {
		belongsTo(name = "buildings", joinType="OUTER");
		hasMany(name = "classroomequipments");
		hasMany(name = "moduleclassrooms");
	}
}

If in controller i use:
building=model("building").findAll(include="classrooms");

I can view/get any column by (for example):
writeDump(var="#building.createdat#");
writeDump(var="#building.classroomcreatedat#");
writeDump(var="#building.size#"); //size is a column of classrooms table

But if i use:
`building=model("building").findAll(select="buildings.*,classrooms.*,CONCAT(buildings.name,' - ',buildings.address) AS building_info", include="classrooms");`

I can view/get all "building" column and only univoque "classrooms" column:
writeDump(var="#building.createdat#");
writeDump(var="#building.size#"); //size is a column of classrooms table

if i try to view/get "classroom" or "classrooms" column with same name as "building" i get always an error:
this NOT work: writeDump(var="#building.classroomname#");
this NOT work: writeDump(var="#building.classroomsname#");
this NOT work: writeDump(var="#building.classroom.name#");
this NOT work: writeDump(var="#building.classrooms.name#");

So how can i get (for example) "classroom" (or "classrooms") column "name"?
Because with #building.name# i get always the name of building, not classroom name.

It is a bug or is the normal logic used by framework?

Thanks

**Comment by mikegrogan** [comment]
You should probably create building_info in your building model like this:

property(name="building_info",sql="CONCAT(buildings.name,' - ',buildings.address)");

**Comment by blastiko** [comment]
Sorry, i did not know this function...
Now is more clear for me, and work ;)
Thank you mikegrogan

**Comment by zainforbjs** [comment]
@blastiko If your question is answered then can you mark this as answered?
